### PR TITLE
XSUP-27024/exabeam

### PR DESCRIPTION
**Pull request by sapirshuker**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/XSUP-27074
## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by sapirshuker** [comment]
Commit number sent is: b91747e

File: Packs/Exabeam/Integrations/Exabeam/Exabeam.py [hunk 1]
```diff
@@ -103,7 +103,7 @@ def test_module_request(self):
         """
         self._http_request('GET', full_url=f'{self._base_url}/api/auth/check', resp_type='text')
 
-    def get_notable_users_request(self, api_unit: str = None, num: str = None, limit: int = None) -> Dict:
+    def get_notable_users_request(self, api_unit: str = None, num: str = None, limit: int = None) -> dict:
```
Comment (AradCarmi) [inline Packs/Exabeam/Integrations/Exabeam/Exabeam.py:106]: Did you change it or the pre-commit hook?

File: Packs/Exabeam/Integrations/Exabeam/Exabeam.yml [hunk 2]
```diff
@@ -84,6 +84,12 @@ configuration:
   name: first_fetch
   type: 0
   required: false
+- display: 'Advanced: Minutes to look back when fetching'
+  name: look_back
+  type: 0
+  required: false
+  defaultvalue: 1
+  additionalinfo: Use this parameter to determine how long backward to look in the search for incidents that were created before the last run time and did not match the query when they were created.
```
Comment (AradCarmi) [inline Packs/Exabeam/Integrations/Exabeam/Exabeam.yml:92]: Add also the limitation if there is one (max lookback)

File: Packs/Exabeam/ReleaseNotes/2_2_7.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+
+#### Integrations
+
+##### Exabeam
+
+- Updated the Docker image to: *demisto/python3:3.10.13.72123*.
+- Added a parameter *Minutes to look back when fetching*.
```
Comment (AradCarmi) [inline Packs/Exabeam/ReleaseNotes/2_2_7.md:7]:
```suggestion
- Added support for the *look-back* parameter in ***fetch-incidents***.
```

**Review by AradCarmi — CHANGES_REQUESTED** [review]
Good job!

File: Packs/Exabeam/Integrations/Exabeam/Exabeam.py (hunk 1 above)
Comment (sapirshuker) [reply Packs/Exabeam/Integrations/Exabeam/Exabeam.py:106]: pre-commit

File: Packs/Exabeam/Integrations/Exabeam/Exabeam.yml (hunk 2 above)
Comment (sapirshuker) [reply Packs/Exabeam/Integrations/Exabeam/Exabeam.yml:92]: The API documentation did not contain any limitations

File: Packs/Exabeam/Integrations/Exabeam/Exabeam.py (hunk 1 above)
Comment (sapirshuker) [reply Packs/Exabeam/Integrations/Exabeam/Exabeam.py:106]: pre-commit

**Review by AradCarmi — APPROVED** [review]

File: Packs/Exabeam/Integrations/Exabeam/README.md [hunk 4]
```diff
@@ -38,6 +38,7 @@ For additional information, refer to [Exabeam Administration Guide](https://docs
     | Fetch incidents |  | False |
     | Max incidents per fetch |  | False |
     | First fetch timestamp (&lt;number&gt; &lt;time unit&gt;, e.g., 12 hours, 7 days) |  | False |
+    | Advanced: Minutes to look back when fetching | Use this parameter to determine how long backward to look in the search for incidents that were created before the last run time and did not match the query when they were created. | False |
```
Comment (eyalpalo) [inline Packs/Exabeam/Integrations/Exabeam/README.md:41]:
```suggestion
    | Advanced: Minutes to look back when fetching | Use this parameter to determine how long backward to look in the search for incidents that were created before the last run time and did not match the query when they were created. Default is 1. | False |
```

**Review by eyalpalo — CHANGES_REQUESTED** [review]
Do we already have a unit test for fetch incidents? If so, add relevant case, if not add unit test.

File: Packs/Exabeam/Integrations/Exabeam/Exabeam_test.py [hunk 5]
```diff
@@ -454,3 +459,56 @@ def test_validate_authentication_params(mocker, args, expected_results):
...
+        )
+    ]
+)
+@freeze_time(datetime(2022, 12, 22, 10, 0, 5, tzinfo=pytz.timezone('UTC')))
+def test_fetch_incidents_with_look_back(mocker, params, expected_incidents, expected_last_run):
+    """
+        Given:
+            - A last run with an incident.
+        When:
+            - executing fetch function with look back.
```
Comment (eyalpalo) [inline Packs/Exabeam/Integrations/Exabeam/Exabeam_test.py:480]:
```suggestion
            - executing fetch_incidents with look back.
```

File: Packs/Exabeam/Integrations/Exabeam/Exabeam_test.py [hunk 6]
```diff
@@ -454,3 +459,56 @@ def test_validate_authentication_params(mocker, args, expected_results):
...
+
+    assert last_run['limit'] == expected_last_run['first_fetch']['limit']
+    assert last_run['time'] == expected_last_run['first_fetch']['time']
+    for id_ in expected_last_run['first_fetch']['found_incident_ids']:
+        assert id_ in last_run['found_incident_ids']
+
+    assert request_get_incidents.call_args_list[0][0][0] == EXPECTED_CALL_ARGS_FOR_LOOK_BACK
+    mocker.patch('Exabeam.demisto.getLastRun', return_value=last_run)
+    request_get_incidents = mocker.patch.object(client, 'get_incidents',
+                                                return_value=INCIDENTS_FOR_LOOK_BACK_SECOND_TIME)
```
Comment (eyalpalo) [inline Packs/Exabeam/Integrations/Exabeam/Exabeam_test.py:508]:
```suggestion
    mocker.patch.object(client, 'get_incidents', return_value=INCIDENTS_FOR_LOOK_BACK_SECOND_TIME)
```

File: Packs/Exabeam/Integrations/Exabeam/Exabeam_test.py [hunk 7]
```diff
@@ -454,3 +459,56 @@ def test_validate_authentication_params(mocker, args, expected_results):
...
+            - A last run with an incident.
+        When:
+            - executing fetch function with look back.
+        Then:
+            - Ensure that the last run time is set correctly.
+            - Check that the query with look-back is set correctly.
+            - Ensure that incidents are correctly filtered.
+    """
+    mocker.patch.object(Client, '_login', return_value=None)
+    client = Client(base_url='https://example.com', username='test_user', password='1234', verify=False, proxy=False,
```
Comment (eyalpalo) [inline Packs/Exabeam/Integrations/Exabeam/Exabeam_test.py:487]: let's break into multiple lines of the Client assignment in order to match pep8 (https://peps.python.org/pep-0008/#introduction) guide.

File: Packs/Exabeam/Integrations/Exabeam/test_data/response_incidents.py [hunk 8]
```diff
@@ -407,7 +533,7 @@
             'medium'
         ],
         'createdAt': [
-            '1671717185195', '1671976385145'
+            '1671717185195', '1671976385145'  # 22/12/2022 13:53:05  25/12/2022 13:53:05
```
Comment (eyalpalo) [inline Packs/Exabeam/Integrations/Exabeam/test_data/response_incidents.py:536]: Uncomment or remove.

**Review by eyalpalo — APPROVED** [review]
Great job! last few comments

**Pull request merged**